### PR TITLE
 Sanitize name of CourseAttributes on rendering instead of before saving #7262 

### LINKED ISSUE #7262: Sanitize name of CourseAttributes on rendering instead of before saving
As mentioned in #7260.

### PR DESCRIPTION
**Pull request by leeyimin**
Fixes #7262 

**Outline of Solution**

* Removed sanitization of name in CourseAttributes
* Sanitize output of course name or course name generated values on front end
* Desanitize course name at appropriate locations to ensure compatibility with existing CourseAttributes with sanitized course name
* Test written to ensure data is properly sanitized.

### CODE REVIEW

**Comment by leeyimin** [comment]
The ordering of the commits are a bit messed up and 2ae45aa is the current last commit of the branch 😅

File: src/main/java/teammates/common/datatransfer/CourseSummaryBundle.java [hunk 1]
```diff
@@ -33,10 +34,13 @@ public int compare(CourseSummaryBundle obj1, CourseSummaryBundle obj2) {
...
      */
     public static void sortSummarizedCoursesByCourseName(List<CourseSummaryBundle> courses) {
+        //TODO: [CourseAttribute] remove desanitization after data migration
+        //desanitization is applied to course name to ensure a well-defined order of the courses by course name
         Collections.sort(courses, new Comparator<CourseSummaryBundle>() {
             @Override
             public int compare(CourseSummaryBundle obj1, CourseSummaryBundle obj2) {
-                return obj1.course.getName().compareTo(obj2.course.getName());
+                return SanitizationHelper.desanitizeIfHtmlSanitized(obj1.course.getName())
+                        .compareTo(SanitizationHelper.desanitizeIfHtmlSanitized(obj2.course.getName()));
```
Comment (thenaesh) [inline src/main/java/teammates/common/datatransfer/CourseSummaryBundle.java]:
```java
String desanitizedCourseName1 = SanitizationHelper.desanitizeIfHtmlSanitized(obj1.course.getName());
String desanitizedCourseName2 = SanitizationHelper.desanitizeIfHtmlSanitized(obj2.course.getName());
return desanitizedCourseName1.compareTo(desanitizedCourseName2);
```

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 2]
```diff
@@ -265,6 +266,93 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
...
+            if (email.getRecipient().equals(student1.email)) {
+                verifyEmail(email, student1.email, subject,
+                        "/sessionClosedEmailTestingSanitizationForStudent.html");
+                hasStudent1ReceivedEmailCorrectly = true;
+            } else if (email.getRecipient().equals(instructor1.email)) {
+                verifyEmail(email, instructor1.email, subject,
+                        "/sessionClosedEmailTestingSanitizationForInstructor.html");
+                hasInstructor1ReceivedEmailCorrectly = true;
+            }
+        }
```
Comment (damithc) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: There is a very similar for-loop a few lines up. Any chance of extracting as a utility method?

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 3]
```diff
@@ -265,6 +266,93 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
...
+        assertTrue(hasStudent1ReceivedEmailCorrectly);
+
+        ______TS("feedback session submission email: sanitization required");
+
+        Calendar time = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        time.set(Calendar.DATE, 4);
+        time.set(Calendar.MONTH, 8);
+        time.set(Calendar.HOUR_OF_DAY, 5);
+        time.set(Calendar.MINUTE, 30);
+        time.set(Calendar.YEAR, 2016);
```
Comment (damithc) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: This code block can be extracted somewhere appropriate as a util method? e.g. 
`Utils.getTime(2016, 4, 8, 5, 30)`

File: src/test/java/teammates/test/cases/browsertests/AdminSearchPageUiTest.java [hunk 4]
```diff
@@ -189,22 +217,14 @@ private boolean isSearchTableHeaderCorrect(int tableNum) {
...
 
-            boolean isStudentCorrect = isStudentContentCorrect(studentRow, student)
-                                       && isStudentLinkCorrect(studentRow, student, instructorToMasquaradeAs);
-
-            if (isStudentCorrect) {
-                return true;
-            }
-        }
+        By by = By.xpath("//table[@id = 'search_table']/tbody/tr[@id='" + AdminSearchPageData.createId(student) + "']");
+        WebElement studentRow = browser.driver.findElement(by);
```
Comment (damithc) [inline src/test/java/teammates/test/cases/browsertests/AdminSearchPageUiTest.java]: Is it possible to push this kind of code into the relevant `*Page` class itself? i.e. the `*Page` object should provide methods to query its contents. Can be a separate PR though.
